### PR TITLE
feat: wire Server B and add multi-project tests and documentation

### PR DESCRIPTION
**Pull request by DevonFulcher**
## Why

Multi-project mode (Server B) needs to register all tool categories so that `DBT_MCP_MULTI_PROJECT_ENABLED=true` provides a fully functional MCP server. The previous phases (1-6) added the individual tool categories; this PR combines them and wires everything together with tests and documentation.

## What

- **Server B wiring**: `server_multiproject.py` now registers all multi-project tools:
  - Multi-project semantic layer (6 `_for_project` variants)
  - Multi-project discovery (18 tools with `config_override`)
  - Multi-project SQL (`text_to_sql_for_project`, `execute_sql_for_project`)
  - Multi-project admin API (`list_jobs_for_project`, `list_jobs_runs_for_project`) including `list_projects` for project discovery
  - Product docs and MCP server metadata (always available)
- **Entry point**: `main.py` calls `create_dbt_mcp_multiproject()` when `DBT_MCP_MULTI_PROJECT_ENABLED=true`
- **Environment resolver**: Extracts prod/dev environment resolution into reusable `resolve_environments()` helper
- **Tests**: Tests covering Server B tool registration, environment resolution, and edge cases
- **Documentation**: README updated with multi-project mode section and tool listings
- **Toolset tracking**: `multiproject_only_tools` set in `toolsets.py` tracks all Server B-exclusive tools for validation

## Notes

- Project discovery on Server B uses the existing `list_projects` tool from the admin API (`register_admin_api_tools`), rather than a separate `list_projects_and_environments` discovery tool. The `Toolset.PROJECT` and `list_projects_and_environments` tool have been removed entirely.
- Admin tools (`list_jobs_for_project`, `list_jobs_runs_for_project`) are registered on both Server A and Server B since they use the shared `register_admin_api_tools()` path
- All 472 unit tests pass, ruff and mypy clean

Drafted by Claude Sonnet 4.6 under the direction of @devonfulcher

### CODE REVIEW

File: src/dbt_mcp/project/environment_resolver.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    page_size: int = 100,
+) -> list[DbtPlatformEnvironmentResponse]:
+    """Fetch all environments for a project using offset/page_size pagination."""
+    offset = 0
+    environments: list[DbtPlatformEnvironmentResponse] = []
+    while True:
+        environments_response = requests.get(
+            f"{dbt_platform_url}/api/v3/accounts/{account_id}/projects/{project_id}/environments/?state=1&offset={offset}&limit={page_size}",
+            headers=headers,
+        )
```
[the remark this answers is not in the document]
Comment (DevonFulcher) [reply src/dbt_mcp/project/environment_resolver.py]: 🤖 _This reply was generated by AI._

This is a false positive. `dbt_platform_url` is a user-configured value from the MCP server's own settings (not from an inbound HTTP request), pointing to the user's own dbt Cloud instance. In this context, the "user" controlling the URL is the same person running the MCP server — there is no untrusted external party that could redirect the request to an unintended host.
